### PR TITLE
[YUNIKORN-324]Add rest API to retrieve cluster resource utilization

### PR DESCRIPTION
**Pull request by HuangTing-Yao**
just return the utilization of memory and vcore.
@yangwwei pls review when you have some time. Thanks.

### CODE REVIEW

**Comment by yangwwei** [comment]
Commented a while back, but I forgot to submit ...

**Comment by kingamarton** [comment]
@HuangTing-Yao  there is an open Jira for refactoring the REST API: https://issues.apache.org/jira/browse/YUNIKORN-413
I am wondering if we can solve this issue according to the new design: https://docs.google.com/document/d/1KgoGqmNGR7TK3yBeqmefiFso2_lQujA_TzlsU0SsG1o/edit#heading=h.9aay2d6eltya
Can you please check if it is possible?
Also can you please add some more unit tests to have the patch coverage check passing?

**Comment by HuangTing-Yao** [comment]
@kingamarton There aren't progress update on yunikorn-413. Perhaps we can done this one first?

**Comment by kingamarton** [comment]
> @kingamarton There aren't progress update on yunikorn-413. Perhaps we can done this one first?

@TaoYang526 I think yes. At least when doing the refactoring this part will already be OK. Let me ping Mani on the Jira to see his opinion.

**Comment by kingamarton** [comment]
> > @kingamarton There aren't progress update on yunikorn-413. Perhaps we can done this one first?
> 
> @TaoYang526 I think yes. At least when doing the refactoring this part will already be OK. Let me ping Mani on the Jira to see his opinion.

@HuangTing-Yao since there is no progress with the refactoring, I think it will OK to follow the old design.

File: pkg/webservice/handlers.go [hunk 1]
```diff
@@ -188,6 +211,31 @@ func getClusterJSON(name string) *dao.ClusterDAOInfo {
...
+		total = 0
+	}
+	used, ok = partition.Root.GetAllocatedResource().Resources[name]
+	if !ok {
+		used = 0
+	}
+	if total < used || total == 0 {
+		percent = int64(0)
+	} else {
+		percent = ((int64(used) * int64(100)) / int64(total))
```
Comment (kingamarton) [inline pkg/webservice/handlers.go:228]: This needs some protection against integer overflow. Both positive and negative. For example if If used = maxInt64, available is 10 then we will have an overflow here. You can find an example in the resources.go fileÉ https://github.com/apache/incubator-yunikorn-core/blob/master/pkg/common/resources/resources.go#L264-L279

Comment (kingamarton) [reply pkg/webservice/handlers.go:228]: Also I think we need to cover the edge cases as well with unit test: nil values, zero values, inf values

**Review by kingamarton — CHANGES_REQUESTED** [review]

File: pkg/webservice/handlers.go (hunk 1 above)
Comment (kingamarton) [reply pkg/webservice/handlers.go:228]: In resources.go file there is already a method for calculating absolute usage: func CalculateAbsUsedCapacity(capacity, used *Resource) *Resource please use that one if possible.

**Review by kingamarton — CHANGES_REQUESTED** [review]

File: pkg/webservice/handlers.go (hunk 1 above)
Comment (HuangTing-Yao) [reply pkg/webservice/handlers.go]: done

**Review by kingamarton — APPROVED** [review]
Thank you @HuangTing-Yao for addressing the comments. LGTM
